py/makeversionhdr.py

```diff
@@ -57,7 +57,7 @@ def get_version_info_from_git():
     return git_tag, git_hash, ver
 
 def get_version_info_from_docs_conf():
-    with open("%s/docs/conf.py" % sys.argv[0].rsplit("/", 2)[0]) as f:
+    with open(os.path.join(os.path.dirname(sys.argv[0]), "docs", "conf.py")) as f:
```

Comment:
argv[0] is some/path/to/micropython/py/makeversionhdr.py so you need to call dirname twice to make it to some/path/to/micropython